### PR TITLE
Stats: logic to determine if Stats v4 is supported

### LINKED ISSUE #1129: Stats: logic to determine if Stats v4 is supported
Create `StatsV4AvailabilityStore` in Yosemite layer to determine if Stats v4 is supported for user by checking if v4 Stats revenue API returns `rest_no_route` error - this will be an async call

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #1129 

## Changes
- Created `StatsV4AvailabilityAction` and `StatsV4AvailabilityStore` for checking if stats v4 is available
  - \+ unit tests with API responses when WC admin was activated and deactivated

## Testing
- CI (no App layer code)

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/StatsV4AvailabilityStore.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Networking
+
+class StatsV4AvailabilityStore: Store {
```
Comment (ctarda) [inline Yosemite/Yosemite/Stores/StatsV4AvailabilityStore.swift]: I think this store can be abstracted to a wider `AvailabilityStore` that potentially we can expand to not only check the availability of stats, but other remote services like plugins or certain modules (i.e. stats) being disabled, so I would suggest renaming it to `AvailabilityStore`.

Also, it should probably be final, unless there is a good reason to potentially subclass it later on?

File: Yosemite/Yosemite/Actions/StatsV4AvailabilityAction.swift [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// MARK: - StatsV4AvailabilityAction: Defines actions regarding Stats v4 availability.
+//
+public enum StatsV4AvailabilityAction: Action {
```
Comment (ctarda) [inline Yosemite/Yosemite/Actions/StatsV4AvailabilityAction.swift]: I think we can abstract this even further and give it a more generic name like  `AvailabilityAction` (see next comment for some more context)

File: Yosemite/Yosemite/Stores/StatsV4AvailabilityStore.swift (hunk 1 above)
Comment (jaclync) [reply Yosemite/Yosemite/Stores/StatsV4AvailabilityStore.swift]: > I think this store can be abstracted to a wider AvailabilityStore that potentially we can expand to not only check the availability of stats, but other remote services like plugins or certain modules (i.e. stats) being disabled, so I would suggest renaming it to AvailabilityStore.

yea this makes sense! will update

> Also, it should probably be final, unless there is a good reason to potentially subclass it later on?

yep will update the class to be final (will make it a default unless there is an immediate plan to subclass)

**Comment by ctarda** [comment]
It seems like Hound is stuck?

**Pull request merged**